### PR TITLE
[5.2] New "preview" mail driver

### PR DESCRIPTION
**Pull request by themsaid**
This PR introduces a new mail driver "Preview", if selected it'll generate a HTML file for each email sent and put it inside a directory specified in the configuration files.

To setup the mail driver a configuration should exist inside `config/mail.php`:

```
'preview' => storage_path('framework/emails'),
```

The generated files will have a format:

```
someone_at_somedomaincom_message_subject.html
```

At the top of every generated file a html comment will be placed:

```
<!--From:me@mail.com, To:someone@somedomain.com, subject:Message Subject-->
```

The scenario is that the developer will open up the .html file in a web browser and see the content of the email as well as the sender, receiver, and subject.

---

**Note 1:** If the specified preview directory doesn't exist, it'll be created along with a `.gitignore` file.
**Note 2:** Generated previews older than 1 minute will be deleted while generating new ones.
**Note 3:** I'm not sure if introducing a new directory inside the storage folder would be considered breaking or not so this PR is a suggestion of a new driver in 5.2 and will be looking for feedback if there are changes to be made.

### CODE REVIEW

File: src/Illuminate/Mail/TransportManager.php [hunk 1]
```diff
@@ -145,6 +146,19 @@ protected function createLogDriver()
...
     /**
+     * Create an instance of the Preview Swift Transport driver.
+     *
+     * @return \Illuminate\Mail\Transport\PreviewTransport
+     */
+    protected function createPreviewDriver()
+    {
+        return new PreviewTransport(
+            $this->app->make('Illuminate\Filesystem\Filesystem'),
+            $this->app->make('config')['mail.preview']
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/TransportManager.php]: please use the same format as everywhere else for reading config

**Comment by taylorotwell** [comment]
Would rather people use something like Mailtrap.io for this.

**Comment by themsaid** [comment]
The idea was to prevent hitting a network activity to check the email output, something like when you're working on regular views and you check the output for tweaks.

Sometimes the internet connection is not that good, so I usually switch to the "log" driver but it doesn't help with visualising the final HTML output.

I think this can be introduced as a package if not a core driver.

**Comment by marktopper** [comment]
@themsaid Make it into a package then, the idea is good.

**Pull request closed without merging**